admin/views/check-form/view-select-forms.ejs

```diff
@@ -40,34 +41,38 @@
               </div>
             </div>
           </th>
+          <% } %>
         </tr>
         </thead>
         <tbody>
-        <% checkFormData.forEach(checkForm => { %>
+        <% checkFormData.forEach((checkForm, i) => { %>
         <tr>
           <td>
             <label for="checkForm<%= checkForm.urlSlug %>">
               <%= checkForm.name %>
             </label>
           </td>
+          <% if (!checkWindowData.isWithinCheckType) { %>
           <td class="align-right">
             <div class="multiple-choice-mtc">
-              <input id="checkForm<%= checkForm.urlSlug %>"
-                     name="checkForm[<%= checkForm.urlSlug %>]"
+              <input id="checkForm-<%= i %>"
+                     name="checkForms[]"
                      type="checkbox"
                      value="<%= checkForm.urlSlug %>"
                      aria-label="Tick form <%= checkForm.name %>"
                      aria-checked="false"
+                     <% if (checkForm.checked) { %>checked="true"<% } %>
                      role="checkbox">
               <div></div>
             </div>
           </td>
         </tr>
+        <% } %>
         <% }) %>
+        <input type="hidden" name="hasAssignedForms" value="<%= hasAssignedForms %>" />
```

Comment:
I wonder why these two hidden variables are required?

Reply:
Both are utilised by the post method.
- `hasAssignedForms` is a convenient way to avoid performing deletion if no check forms have been assigned to the check window at the given state.
- `checkWindowName` is being used to display the name of the check window on the green notification bar. I've done this to avoid an additional data service call in the post method to fetch just that information.

Reply:
If you are using a value from the GUI to determine whether or not you need to run a delete operation, then that's bad as the GUI is entirely untrusted.  Instead, either always run the delete or determine at run time.

In my view, the `checkWindowName` should also be looked up, rather than passed in through the web request.  I don't see any issue with that.